deps/rabbit/src/rabbit_ff_registry_factory.erl

```diff
@@ -171,19 +168,23 @@ initialize_registry(NewSupportedFeatureFlags,
                     NewFeatureStates,
                     WrittenToDisk) ->
     try
-        Ret = maybe_initialize_registry(NewSupportedFeatureFlags,
-                                        NewFeatureStates,
-                                        WrittenToDisk),
-        case Ret of
-            ok      -> ok;
-            restart -> initialize_registry(NewSupportedFeatureFlags,
-                                           NewFeatureStates,
-                                           WrittenToDisk);
-            Error1  -> Error1
-        end
+        true = global:set_lock(?FF_REGISTRY_LOADING_LOCK, [node()]),
+        ?LOG_DEBUG(
+           "Feature flags: acquired lock before initializing registry (~tp)",
+           [self()],
+           #{domain => ?RMQLOG_DOMAIN_FEAT_FLAGS}),
+        ok = maybe_initialize_registry(NewSupportedFeatureFlags,
+                                       NewFeatureStates,
+                                       WrittenToDisk)
     catch
         throw:{error, _} = Error2 ->
             Error2
+    after
+        ?LOG_DEBUG(
+           "Feature flags: released lock after initializing registry (~tp)",
+           [self()],
+           #{domain => ?RMQLOG_DOMAIN_FEAT_FLAGS}),
+        true = global:del_lock(?FF_REGISTRY_LOADING_LOCK, [node()])
```

Comment:
Technically, the lock is not "released" yet at the time of the log message :-)

So either the log message says "release" or it is moved after the unlock.

Reply:
Good point! Swapped to saying "releasing"